linkschkr.go

```diff
@@ -0,0 +1,127 @@
+package linkschkr
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"golang.org/x/net/html"
+)
+
+type Work struct {
+	site   string
+	result chan *Result
+}
+
+type Result struct {
+	site      string
+	up        bool
+	extraURIs []string
+	workerID  int
+}
+
+func Crawler(ID int, c <-chan *Work) {
+	for {
+		work := <-c
+		result := &Result{
+			site:     work.site,
+			up:       true,
+			workerID: ID,
+		}
+		var resp *http.Response
+		var err error
+		// Avoid to download binaries
```

Comment:
Maybe we should always do a HEAD request first, and then only try to GET the resource if its content type is HTML?

Reply:
why should I use the http header if I can add one more switch statement :P
Let's list all binaries types in the world :P